libcloudforensics/providers/aws/internal/ec2.py

```diff
@@ -137,3 +142,364 @@ def AttachVolume(self,
+          instance_id = instance['InstanceId']
+          aws_instance = AWSInstance(
+              self.aws_account, instance_id, zone[:-1], zone)
+
+          for tag in instance.get('Tags', []):
+            if tag.get('Key') == 'Name':
+              aws_instance.name = tag.get('Value')
+              break
+
+          instances[instance_id] = aws_instance
+    return instances
+
+  def GetInstancesByNameOrId(
+      self,
+      instance_name: Optional[str] = None,
+      instance_id: Optional[str] = None,
+      region: Optional[str] = None) -> List[AWSInstance]:
+    """Get instances from an AWS account by their name tag or an ID.
+
+    Exactly one of [instance_name, instance_id] must be specified. If looking up
+    an instance by its ID, the method returns a list with exactly one
+    element. If looking up instances by their name tag (which are not unique
+    across instances), then the method will return a list of all instances
+    with that name tag, or an empty list if no instances with matching name
+    tag could be found.
+
+    Args:
+      instance_name (str): Optional. The instance name tag of the instance to
+          get.
+      instance_id (str): Optional. The instance id of the instance to get.
+      region (str): Optional. The region to look the instance in.
+          If none provided, the default_region associated to the AWSAccount
+          object will be used.
+
+    Returns:
+      List[AWSInstance]: A list of Amazon EC2 Instance objects.
+
+    Raises:
+      ValueError: If both instance_name and instance_id are None or if both
+          are set.
+    """
+
+    if (not instance_name and not instance_id) or (instance_name and instance_id):  # pylint: disable=line-too-long
+      raise ValueError('You must specify exactly one of [instance_name, '
+                       'instance_id]. Got instance_name: {0:s}, instance_id: '
+                       '{1:s}'.format(str(instance_name), str(instance_id)))
+    if instance_name:
+      return self.GetInstancesByName(instance_name, region=region)
+    assert instance_id  # Mypy: assert that instance_id is not None
+    return [self.GetInstanceById(instance_id, region=region)]
+
+  def GetInstancesByName(self,
+                         instance_name: str,
+                         region: Optional[str] = None) -> List[AWSInstance]:
+    """Get all instances from an AWS account with matching name tag.
+
+    Args:
+      instance_name (str): The instance name tag.
+      region (str): Optional. The region to look the instance in.
+          If none provided, the default_region associated to the AWSAccount
+          object will be used.
+
+    Returns:
+      List[AWSInstance]: A list of EC2 Instance objects. If no instance with
+          matching name tag is found, the method returns an empty list.
+    """
+
+    instances = self.ListInstances(region=region)
+    return [instance for _, instance in instances.items() if
+            instance.name == instance_name]
```

Comment:
use `values()` instead of `items()` if you don't care about the keys
```suggestion
    return [instance for instance in instances.values() if
            instance.name == instance_name]
```

Reply:
Done